src/test/java/org/tarantool/jdbc/JdbcPreparedStatementIT.java

```diff
@@ -477,6 +489,151 @@ public void testExecuteStringBatchQuery() throws Exception {
         assertThrows(SQLException.class, () -> prep.addBatch("INSERT INTO test(id, val) VALUES (1, 'one')"));
     }
 
+    @Test
+    public void testSetAsciiStream() throws Exception {
+        prep = conn.prepareStatement("INSERT INTO test(id, val) VALUES (?, ?)");
+        prep.setInt(1, 1);
+        InputStream asciiStream = new ByteArrayInputStream("one".getBytes("ASCII"));
+        prep.setAsciiStream(2, asciiStream);
+
+        assertFalse(prep.execute());
+        assertEquals("one", consoleSelect(1).get(1));
+    }
+
+    @Test
+    public void testSetAsciiLimitedStream() throws Exception {
+        prep = conn.prepareStatement("INSERT INTO test(id, val) VALUES (?, ?)");
+        prep.setInt(1, 1);
+        InputStream asciiStream = new ByteArrayInputStream("one and two and even three".getBytes("ASCII"));
+        prep.setAsciiStream(2, asciiStream, 3);
+
+        assertFalse(prep.execute());
+        assertEquals("one", consoleSelect(1).get(1));
+    }
+
+    @Test
+    public void testSetNegativeAsciiStream() throws Exception {
+        prep = conn.prepareStatement("INSERT INTO test(id, val) VALUES (?, ?)");
+        prep.setInt(1, 1);
+        InputStream asciiStream = new ByteArrayInputStream("one and two and even three".getBytes("ASCII"));
+        SQLException error = assertThrows(SQLException.class, () -> prep.setAsciiStream(2, asciiStream, -10));
+        assertEquals(SQLStates.INVALID_PARAMETER_VALUE.getSqlState(), error.getSQLState());
+    }
+
+    @Test
+    public void testSetBadStream() throws Exception {
+        prep = conn.prepareStatement("INSERT INTO test(id, val) VALUES (?, ?)");
+
+        InputStream throwingStream = mock(InputStream.class);
+        when(throwingStream.read(anyObject(), anyInt(), anyInt())).thenThrow(IOException.class);
+
+        SQLException error = assertThrows(
+            SQLException.class,
+            () -> prep.setAsciiStream(2, throwingStream)
+        );
+        assertEquals(SQLStates.INVALID_PARAMETER_VALUE.getSqlState(), error.getSQLState());
```

Comment:
Maybe it worth to check that a cause is the same as the exception that was raised inside a stream?